### PR TITLE
RFR - Add search functionality and feature test

### PR DESCRIPTION
**Pull request by kaylarr**

### CODE REVIEW

File: app/controllers/searches_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if category_id.blank?
+      @plans = Plan.all
+    else
+      @plans = Plan.where(category_id: category_id)
+    end
+
+    # Filter by budget range
+    budget = params[:search][:'search_budget']
+
+    if !budget.blank?
```
Comment (davidrf) [inline app/controllers/searches_controller.rb]: unless budget.blank?

File: app/controllers/searches_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      min = budget.to_i * 0.9
+      max = budget.to_i * 1.1
+
+      @plans = @plans.where("budget > ?::money AND budget < ?::money", min, max)
+    end
+
+    # Filter by inclusion of keywords
+    keywords = params[:search][:keywords].split
+    if !keywords.blank?
```
Comment (davidrf) [inline app/controllers/searches_controller.rb]: `unless keywords.blank?`

File: app/views/searches/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<h1>Search Results</h1>
+
+<% if !@plans.nil? %>
```
Comment (davidrf) [inline app/views/searches/index.html.erb]: `unless @plans.blank?`

File: app/controllers/searches_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class SearchesController < ApplicationController
+  def index
```
Comment (davidrf) [inline app/controllers/searches_controller.rb]: Your controllers should be skinny. This controller should be a one liner.
```ruby
def index
  @plans = Plan.search_result(params)
end
```
`#search_result` is a class method on your Plan model. I would declare category_id, budget, and keywords, as you have here, and then put the filter by category  as you have here. Then I would have two more class methods (`#budget_filter`, and `#keyword_filter` that would only be run on your @plans if there is a budget and keyword specified.

Comment (kaylarr) [inline app/controllers/searches_controller.rb]: Done - all logic moved to Plan class functions

**Pull request merged**